ladybug_geometry/geometry2d/arc.py

```diff
@@ -0,0 +1,400 @@
+
+    @classmethod
+    def from_start_mid_end(cls, p1, m, p2, circle=False):
+        """Initialize a new arc from start, middle, and end points.
+
+        Note that input points will be assumed to be in counterclockwise order.
+
+        Args:
+            p1: The start point of the arc.
+            m: Any point along the length of the arc that is not the start or end.
+            p2: The end point of the arc.
+            circle: Set to True if you would like the output to be a full circle
+                defined by the three points instead of an arc with a start and end.
+                Default is False.
+        """
+        for pt in (p1, m, p2):
+            assert isinstance(pt, Point2D), "Expected Point2D. Got {}.".format(type(pt))
+        e1 = (p1.x ** 2 + p1.y ** 2)
+        e2 = (m.x ** 2 + m.y ** 2)
+        e3 = (p2.x ** 2 + p2.y ** 2)
+        den = 2 * (p1.x * (m.y - p2.y) - p1.y * (m.x - p2.x) + m.x * p2.y - p2.x * m.y)
+        try:
+            x = -(e1 * (p2.y - m.y) + e2 * (p1.y - p2.y) + e3 * (m.y - p1.y)) / den
+            y = -(e1 * (m.x - p2.x) + e2 * (p2.x - p1.x) + e3 * (p1.x - m.x)) / den
+        except ZeroDivisionError:
+            raise ValueError('Input points {}, {}, {} are colinear and '
+                             'cannot define an arc.'.format(p1, m, p2))
+        r = math.sqrt((x - p1.x) ** 2 + (y - p1.y) ** 2)
+        if circle is True:
+            return cls(Point2D(x, y), r)
+        else:
+            a1 = Vector2D(1, 0).angle_counterclockwise(Vector2D(p1.x - x, p1.y - y))
+            a2 = Vector2D(1, 0).angle_counterclockwise(Vector2D(p2.x - x, p2.y - y))
+            return cls(Point2D(x, y), r, a1, a2)
+
+    @property
+    def c(self):
+        """Center point of the circle on which the arc lies."""
+        return self._c
+
+    @property
+    def r(self):
+        """Radius of arc."""
+        return self._r
+
+    @property
+    def a1(self):
+        """Start angle of the arc in radians."""
+        return self._a1
+
+    @property
+    def a2(self):
+        """End angle of the arc in radians."""
+        return self._a2
+
+    @property
+    def p1(self):
+        """Start point."""
+        return Point2D(
+            self.c.x + self._cos_a1 * self.r, self.c.y + self._sin_a1 * self.r)
+
+    @property
+    def p2(self):
+        """End point."""
+        return Point2D(
+            self.c.x + self._cos_a2 * self.r, self.c.y + self._sin_a2 * self.r)
+
+    @property
+    def midpoint(self):
+        """Midpoint."""
+        return self.point_at(0.5)
+
+    @property
+    def length(self):
+        """The length of the arc."""
+        return self.angle * self.r
+
+    @property
+    def angle(self):
+        """The total angle over the domain of the arc in radians."""
+        _diff = self._a2 - self._a1
+        return _diff if not self.is_inverted else 2 * math.pi + _diff
+
+    @property
+    def area(self):
```

Comment:
shouldn't we only get the area if the arc is a circle? I know that you mentioned what the area is in docstring but I suspect people will make mistake and use this area for open arcs unless we raise an exception.

Reply:
Ok, I can do that. I'll raise an exception.